### PR TITLE
constexpr DEBUG_ASSERT

### PR DESCRIPTION
**Pull request by daschuer**
Replace the unused conditional assignment by plain if() to make our asserts usable in contsexpr

In case the assertion failed the compiler is already complaining that the assertion is no longer const, because QCritical is not const.

### CODE REVIEW

**Comment by daschuer** [comment]
Oh, it turns out we can remove the lambdas completely. 
Done.

**Review by uklotzde — APPROVED** [review]

File: src/util/assert.h [hunk 1]
```diff
@@ -52,11 +52,20 @@ inline void mixxx_release_assert(const char* assertion, const char* file, int li
 //
 // In release builds, doSomething() is never called!
 #ifdef MIXXX_DEBUG_ASSERTIONS_ENABLED
-#define DEBUG_ASSERT(cond) ((!(cond)) ? mixxx_debug_assert(#cond, __FILE__, __LINE__, ASSERT_FUNCTION) : mixxx_noop())
+#define DEBUG_ASSERT(cond)                                              \
+    if (Q_UNLIKELY(!(cond))) {                                          \
```
Comment (uklotzde) [inline src/util/assert.h:56]: How about replacing `(cond)` with `static_cast<bool>(cond)` as suggested by @Holzhaus?

Comment (daschuer) [reply src/util/assert.h]: Done

**Comment by daschuer** [comment]
Unfortunately the error message is a bit weird in case of a compile time assert: 
```
home/sperry/workspace/2.3/src/mixxxapplication.cpp: In constructor ‘MixxxApplication::MixxxApplication(int&, char**)’:
/home/sperry/workspace/2.3/src/mixxxapplication.cpp:79:28:   in ‘constexpr’ expansion of ‘{anonymous}::makei(0)’
/home/sperry/workspace/2.3/src/util/assert.h:57:27: error: call to non-‘constexpr’ function ‘void mixxx_debug_assert(const char*, const char*, int, const char*)’
   57 |         mixxx_debug_assert(#cond, __FILE__, __LINE__, ASSERT_FUNCTION); \
      |         ~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/sperry/workspace/2.3/src/mixxxapplication.cpp:60:5: note: in expansion of macro ‘DEBUG_ASSERT’
   60 |     DEBUG_ASSERT(i);
      |     ^~~~~~~~~~~~
```
that is also the case with the std assert(), not our fault

**Comment by uklotzde** [comment]
@Holzhaus Merge?

**Comment by Holzhaus** [comment]
Didn't test this yet, but feel free to merge if you did. Otherwise I can have a look in the next days.

**Comment by uklotzde** [comment]
LGTM

**Pull request merged**